styleguide_example/blog_examples/google_login_server_flow/raw/service.py

```diff
@@ -0,0 +1,125 @@
+from random import SystemRandom
+from typing import Any, Dict, Tuple
+from urllib.parse import urlencode
+
+import jwt
+import requests
+from attr import define
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured, ValidationError
+from django.urls import reverse_lazy
+from oauthlib.common import UNICODE_ASCII_CHARACTER_SET
+
+
+@define
+class GoogleRawLoginCredentials:
+    client_id: str
+    client_secret: str
+
+
+class GoogleRawLoginFlowService:
+    API_URI = reverse_lazy("api:google-oauth2:login-raw:callback-raw")
+
+    GOOGLE_AUTH_URL = "https://accounts.google.com/o/oauth2/v2/auth"
+    SCOPES = [
+        "https://www.googleapis.com/auth/userinfo.email",
+        "https://www.googleapis.com/auth/userinfo.profile",
+        "openid",
+    ]
+
+    GOOGLE_ID_TOKEN_INFO_URL = "https://www.googleapis.com/oauth2/v3/tokeninfo"
+    GOOGLE_ACCESS_TOKEN_OBTAIN_URL = "https://oauth2.googleapis.com/token"
+    GOOGLE_USER_INFO_URL = "https://www.googleapis.com/oauth2/v3/userinfo"
+
+    def __init__(self):
+        self._credentials = google_raw_login_get_credentials()
+
+    @staticmethod
+    def _jwt_decode_token(token: str) -> Dict[str, Any]:
```

Comment:
We can remove this method and just inline the code in the `decode_id_token` method :+1: